### PR TITLE
feat: Extend the JenkinsClient to include the `fetchBuildByNumber` method

### LINKED ISSUE #1285: CI Integrations: Extend the JenkinsClient to include the `fetchBuildByNumber` method
**Description**
Extend the JenkinsClient to include the `fetchBuildByNumber` method.

**Acceptance Criteria**
- [ ] Add `.fetchBuildByNumber` method to the JenkinsClient.
- [ ] Test new functionality using the JenkinsMockServer.

### PR DESCRIPTION
**Pull request by solid-daniyilyevtyushkin**
Fixes #1285

## Checklist

- [x] Documented
- [x] Unit tested

### CODE REVIEW

File: metrics/ci_integrations/test/test_utils/matchers.dart [hunk 1]
```diff
@@ -4,7 +4,11 @@
 import 'package:ci_integration/cli/error/sync_error.dart';
 import 'package:test/test.dart';
 
-/// A matcher that can be used to detect that test case throws
+/// A matcher that can be used to detect that a test case throws
+/// an [AssertionError].
+final Matcher throwsAssertionError = throwsA(isA<AssertionError>());
```
Comment (solid-vladlaba) [inline metrics/ci_integrations/test/test_utils/matchers.dart:9]: Good idea to simplify tests - thanks!

File: metrics/ci_integrations/lib/client/jenkins/jenkins_client.dart [hunk 2]
```diff
@@ -280,6 +280,30 @@ class JenkinsClient with LoggerMixin {
...
+    int buildNumber,
+  ) {
+    logger.info('Fetching a #$buildNumber build of a $jobName job...');
+
+    assert(jobName != null);
+    assert(buildNumber != null);
+
+    final jobPath = _jobFullNameToPath(jobName);
+
+    final url = _jenkinsUrlBuilder.build(
```
Comment (solid-vladlaba) [inline metrics/ci_integrations/lib/client/jenkins/jenkins_client.dart:299]:
```suggestion
    final jobPath = _jobFullNameToPath(jobName);
    final url = _jenkinsUrlBuilder.build(
```

**Review by solid-vladlaba — CHANGES_REQUESTED** [review]
Well done,  @solid-daniyilyevtyushkin @solid-yevheniiashchypeleva! I've added a couple of suggestions below - please let me know what you think! Also, @solid-vovabeloded will finish the review for this PR.

**Review by solid-vovabeloded — CHANGES_REQUESTED** [review]
Great job, @solid-daniyilyevtyushkin & @solid-yevheniiashchypeleva! I've added a couple of suggestions here - please take a look. Overall, the code looks great - thanks!

**Review by solid-vovabeloded — CHANGES_REQUESTED** [review]
Almost there, @solid-daniyilyevtyushkin & @solid-yevheniiashchypeleva! I've added a bit more minor suggestions below - let me know what do you think. Overall, the code looks great - thanks!

**Review by solid-vovabeloded — APPROVED** [review]
LGTM, thanks @solid-daniyilyevtyushkin & @solid-yevheniiashchypeleva!

**Review by solid-vladlaba — APPROVED** [review]
LGTM, thanks @solid-daniyilyevtyushkin @solid-yevheniiashchypeleva!:rocket: 
And @solid-vovabeloded for review! 🙂

**Pull request merged**